src/graft_dag.hpp

```diff
@@ -0,0 +1,178 @@
+// Copyright 2019-2022 bito project contributors.
+// bito is free software under the GPLv3; see LICENSE file for details.
+//
+// The GraftDAG is a proposed addition (graft) to SubsplitDAG (host), which we
+// can perform computations on without the need for adding nodes and edges and
+// reindexing the full DAG.
+// NOTE: This may add overhead to DAG traversals.
+
+#include "gp_dag.hpp"
+#include "quartet_hybrid_request.hpp"
+#include "subsplit_dag.hpp"
+
+#pragma once
+
+class GPDAG;
+
+class GraftDAG {
+ public:
+  // ** Constructors:
+
+  // Initialize empty GraftDAG.
+  GraftDAG(SubsplitDAG &dag);
+  // Initialize GraftDAG with an initial graft.
+  GraftDAG(SubsplitDAG &dag, NNIOperation &nni);
+  GraftDAG(SubsplitDAG &dag, Bitset &parent_subsplit, Bitset &child_subsplit);
+
+  // ** Comparators
+
+  // Uses same method of comparison as SubsplitDAG (node and edge sets).
+  static int Compare(GraftDAG &dag_a, GraftDAG &dag_b);
+  // Treats GraftDAG as completed SubsplitDAG to compare against normal
+  // SubsplitDAG.
+  static int CompareToDAG(GraftDAG &dag_a, GPDAG &dag_b);
```

Comment:
I don't think that we need this to be a GPDAG-- it can be just a sDAG, right?

Reply:
Fixed to SubsplitDAG.